Fix compareplot input df column mismatch after we adapted arviz

junpenglao opened the pull request:
Close #3401.

junpenglao committed:
Fix #3401

canyon289 commented on pymc3/plots/__init__.py:
```diff
@@ -53,6 +52,38 @@ def wrapped(*args, **kwargs):
+                                          'dSE': 'dse',
+                                          'shape_warn': 'warning'})
+    args[0] = comp_df
+    args = tuple(args)
```
Why does args need to be coerced into a tuple? The list can be unpacked in line 85 as well right?

canyon289 commented on pymc3/plots/__init__.py:
```diff
@@ -53,6 +52,38 @@ def wrapped(*args, **kwargs):
+@functools.wraps(az.plot_compare)
+def compareplot(*args, **kwargs):
+    args = list(args)
+    comp_df = args[0].copy()
```
This assumes that comp_df is passed as the 0th arg. Does this cover the case where comp_df is passed a kwarg?

junpenglao committed:
address comments

junpenglao committed:
remove unnecessary mapping

compareplot is model level (i.e., no need to map between varnames and var_names kwarg)

junpenglao commented:
@canyon289 any other comments? Otherwise will merge.

canyon289 commented:
Sorry give me three hours and I'll do final review. Apologies on delay

On Monday, March 11, 2019, Junpeng Lao <notifications@github.com> wrote:

> @canyon289 <https://github.com/canyon289> any other comments? Otherwise
> will merge.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pymc-devs/pymc3/pull/3402#issuecomment-471748347>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AG4S4dybGz6AnbLhVKdSdw1V-dOPUNv1ks5vVs_9gaJpZM4bnjoX>
> .
>

canyon289 commented:
This looks fine to me, especially since its temporary code that is going to get removed in a couple releases!
